### PR TITLE
[SYCL] Add note on CGF evaluation

### PR DESCRIPTION
**Pull request by Bensuo**
- Add a note on CGF evaluation behaviour.

### CODE REVIEW

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc [hunk 1]
```diff
@@ -1070,6 +1070,15 @@ device at once.
 Host tasks can be updated as part of <<executable-graph-update, executable graph update>>
 by replacing the whole node with the new callable.
 
+=== Command Group Function Evaluation
+
+Host code within a command group function object is evaluated once either when
+submitting a command to a queue in recording mode or when adding a node to a
+graph explicitly. This behaviour is in keeping with the SYCL 2020 specification
```
Comment (EwanC) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc]: A link to the relevant SYCL 2020 spec section would be nice here.

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc [hunk 2]
```diff
@@ -1070,6 +1070,15 @@ device at once.
 Host tasks can be updated as part of <<executable-graph-update, executable graph update>>
 by replacing the whole node with the new callable.
 
+=== Command Group Function Evaluation
+
+Host code within a command group function object is evaluated once either when
```
Comment (EwanC) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc]: Would reword this slightly to say the generic behaviour, then give the two graph constructing situations:

Host code within a command group function object is evaluated ~~once~~ **when the command-group is added to the graph**, this is either when ...

**Review by EwanC — APPROVED** [review]

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc (hunk 2 above)
Comment (reble) [reply sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc]: Agreed, and we might want to exclude a `host_task` explicitly here. It could be seen as host code.

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc (hunk 1 above)
Comment (Bensuo) [reply sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc]: Done!

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc (hunk 2 above)
Comment (Bensuo) [reply sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc]: I've reworded as suggested and also added a note that it does not apply to `host_task` along with a short code example to help illustrate.

**Review by EwanC — APPROVED** [review]
Nice example 😃

**Review by reble — APPROVED** [review]
